x-pack/plugins/fleet/common/types/models/epm.ts

```diff
@@ -269,13 +270,24 @@ export interface RegistryDataStream {
   [RegistryDataStreamKeys.ingest_pipeline]: string;
   [RegistryDataStreamKeys.elasticsearch]?: RegistryElasticsearch;
   [RegistryDataStreamKeys.dataset_is_prefix]?: boolean;
+  [RegistryDataStreamKeys.permissions]?: RegistryDataStreamPermissions;
 }
 
 export interface RegistryElasticsearch {
   'index_template.settings'?: object;
   'index_template.mappings'?: object;
 }
 
+export interface RegistryDataStreamPermissions {
+  cluster?: string[];
+  indices?: string[];
+}
+
+export interface PackagePermissions {
```

Comment:
I'm not sure where to put this or how to call it. I'm open to suggestions

Reply:
I would move this to `models/agent_policy.ts` right under `FullAgentPolicyOutputPermissions`

actually, what do you think of renaming like this?:
`FullAgentPolicyOutputPermissions` -> `FullAgentPolicyOutputRoles`
`PackagePermissions` -> `FullAgentPolicyOutputRolePermissions`